src/components/home/SideBarNavLink.tsx

```diff
@@ -1,34 +1,54 @@
-import { FunctionComponent } from "react";
+import { CSSProperties, FunctionComponent } from "react";
 import { Link, useMatch, useResolvedPath } from "react-router-dom";
-import { Stack } from "@mui/material";
+import { Box, Stack } from "@mui/material";
 import theme from "theme";
 
+const activeBackground = "rgba(255, 255, 255, 0.1)";
+
+interface ButtonProps {
+  readonly onClick?: VoidFunction;
+  readonly style: CSSProperties;
+}
+
+interface LinkProps extends ButtonProps {
+  readonly to: string;
+}
+
+type WrapperProps = LinkProps | ButtonProps;
+
 interface SideBarNavLinkProps {
   readonly label: string;
   readonly icon: JSX.Element;
-  readonly to: string;
-  readonly closeMenu?: VoidFunction;
+  readonly to?: string;
+  readonly onClick?: VoidFunction;
 }
 
-const activeBackground = "rgba(255, 255, 255, 0.1)";
+const Wrapper: FunctionComponent<WrapperProps> = (props) => {
```

Comment:
I think this could cause confusion for someone who hasn't seen this part before but since it's isolated to the sidebar button logic it should be okay.

Reply:
Just to clarify the reason for the separate `box` is because authentication and protected routes already take care of navigation?